packages/block-editor/src/components/segmented-text-control/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { BaseControl, Button } from '@wordpress/components';
+
+/**
+ * Control to facilitate segmented text selections.
+ *
+ * @param {Object}   props           Component props.
+ * @param {string}   props.label     A label for the control.
+ * @param {string}   props.value     Currently selected value.
+ * @param {Function} props.onChange  Callback to handle onChange.
+ * @param {Array}    props.controls  Array of controls to display.
```

Comment:
Would be nice to specify here what attributes each control object should contain.

Reply:
Agreed, something mentioning `value`, `icon`, `label` etc. would be handy.